crypto/rsa/rsa_lib.c

```diff
@@ -733,3 +734,398 @@ int rsa_get0_all_params(RSA *r, STACK_OF(BIGNUM_const) *primes,
 
     return 1;
 }
+
+int EVP_PKEY_CTX_set_rsa_padding(EVP_PKEY_CTX *ctx, int pad_mode)
+{
+    OSSL_PARAM pad_params[2], *p = pad_params;
+
+    if (ctx == NULL) {
+        ERR_raise(ERR_LIB_EVP, EVP_R_COMMAND_NOT_SUPPORTED);
+        /* Uses the same return values as EVP_PKEY_CTX_ctrl */
+        return -2;
+    }
+
+    /* If key type not RSA or RSA-PSS return error */
+    if (ctx->pmeth != NULL
+            && ctx->pmeth->pkey_id != EVP_PKEY_RSA
```

Comment:
What is EVP_PKEY_RSA2 used for?

Reply:
EVP_PKEY_RSA2 represents a legacy OID and is equivalent to EVP_PKEY_RSA. It's only ever used in an `ameth`. The `pmeth` structures only ever have EVP_PKEY_RSA or EVP_PKEY_RSA_PSS.